sdk/eventhub/azure-eventhub/conftest.py

```diff
@@ -66,59 +72,81 @@ def get_logger(filename, level=logging.INFO):
 log = get_logger(None, logging.DEBUG)
 
 
-def create_eventhub(eventhub_config, client=None):
-    hub_name = str(uuid.uuid4())
-    hub_value = EventHub(partition_count=PARTITION_COUNT)
-    client = client or ServiceBusService(
-        service_namespace=eventhub_config['namespace'],
-        shared_access_key_name=eventhub_config['key_name'],
-        shared_access_key_value=eventhub_config['access_key'])
-    if client.create_event_hub(hub_name, hub=hub_value, fail_on_exist=True):
-        return hub_name
-    raise ValueError("EventHub creation failed.")
-
-
-def cleanup_eventhub(eventhub_config, hub_name, client=None):
-    client = client or ServiceBusService(
-        service_namespace=eventhub_config['namespace'],
-        shared_access_key_name=eventhub_config['key_name'],
-        shared_access_key_value=eventhub_config['access_key'])
-    client.delete_event_hub(hub_name)
+@pytest.fixture(scope="session")
+def resource_group():
+    try:
+        SUBSCRIPTION_ID = os.environ["AZURE_SUBSCRIPTION_ID"]
+    except KeyError:
+        pytest.skip('AZURE_SUBSCRIPTION_ID defined')
+        return
+    resource_client = ResourceManagementClient(EnvironmentCredential(), SUBSCRIPTION_ID)
+    resource_group_name = RES_GROUP_PREFIX + str(uuid.uuid4())
+    try:
+        rg = resource_client.resource_groups.create_or_update(
+           resource_group_name, {"location": LOCATION}
+        )
+        yield rg
+    finally:
+        try:
+            resource_client.resource_groups.begin_delete(resource_group_name)
+        except:
+            warnings.warn(UserWarning("resource group teardown failed"))
 
 
 @pytest.fixture(scope="session")
-def live_eventhub_config():
+def eventhub_namespace(resource_group):
     try:
-        config = {}
-        config['hostname'] = os.environ['EVENT_HUB_HOSTNAME']
-        config['event_hub'] = os.environ['EVENT_HUB_NAME']
-        config['key_name'] = os.environ['EVENT_HUB_SAS_POLICY']
-        config['access_key'] = os.environ['EVENT_HUB_SAS_KEY']
-        config['namespace'] = os.environ['EVENT_HUB_NAMESPACE']
-        config['consumer_group'] = "$Default"
-        config['partition'] = "0"
-        config['connection_str'] = CONN_STR
+        SUBSCRIPTION_ID = os.environ["AZURE_SUBSCRIPTION_ID"]
     except KeyError:
-        pytest.skip("Live EventHub configuration not found.")
-    else:
-        return config
+        pytest.skip('AZURE_SUBSCRIPTION_ID defined')
+        return
```

Comment:
(nothing wrong here and no change is required, but I don't think we need to catch KeyError here (and in the live_eventhub fixure) because the resource_group fixture has already done the check for us)

Reply:
I lean to making it explicit in namespace and eh creation unless it harms. Using resource group's skip is a hidden logic.